### PR TITLE
checker: create operator to match leader role

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?

Fix https://github.com/tikv/pd/issues/3130

We only have 'follower' -> 'none follower' logic, no 'none leader' -> 'leader'

### What is changed and how it works?
- add 'none leader' -> 'leader' logic
- update metrics name to distinguish different types of operators
- add test

### Check List

Tests
- Unit test

Related changes
- Need to cherry-pick to the release branch

### Release note
- Fix the issue that the leader role does not take effect when using the replacement rule in some cases

### CODE REVIEW

File: server/schedule/checker/rule_checker.go [hunk 1]
```diff
@@ -138,11 +138,15 @@ func (c *RuleChecker) fixLooseMatchPeer(region *core.RegionInfo, fit *placement.
 		checkerCounter.WithLabelValues("rule_checker", "fix-peer-role").Inc()
 		return operator.CreatePromoteLearnerOperator("fix-peer-role", c.cluster, region, peer)
 	}
-	if region.GetLeader().GetId() == peer.GetId() && rf.Rule.Role == placement.Follower {
+	if region.GetLeader().GetId() != peer.GetId() && rf.Rule.Role == placement.Leader {
 		checkerCounter.WithLabelValues("rule_checker", "fix-leader-role").Inc()
+		return operator.CreateTransferLeaderOperator("fix-leader-role", c.cluster, region, region.GetLeader().StoreId, peer.GetStoreId(), 0)
+	}
```
Comment (Yisaer) [inline server/schedule/checker/rule_checker.go:144]: Basically LGTM. What about add a storeStateFilter?

Comment (HunDunDM) [reply server/schedule/checker/rule_checker.go:144]: Describe the scene?

Comment (disksing) [reply server/schedule/checker/rule_checker.go]: @HunDunDM maybe the store is busy. Please check the new test I add.

**Review by HunDunDM — APPROVED** [review]

**Comment by disksing** [comment]
@Yisaer Please take another look

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**